src/astarte-client/client.ts

```diff
@@ -366,24 +371,19 @@ class AstarteClient {
     );
   }
 
-  async registerDevice({ deviceId, introspection }: any): Promise<any> {
+  async registerDevice(params: {
+    deviceId: AstarteDevice['id'];
+    introspection?: { [interfaceName: string]: AstarteInterfaceDescriptor };
+  }): Promise<{ credentialsSecret: string }> {
+    const { deviceId, introspection } = params;
     const requestBody: any = {
       hw_id: deviceId,
     };
-
     if (introspection) {
-      const encodedIntrospection: any = {};
-      Array.from(introspection).forEach(([key, interfaceDescriptor]: any) => {
-        encodedIntrospection[key] = {
-          major: interfaceDescriptor.major,
-          minor: interfaceDescriptor.minor,
-        };
-      });
-      requestBody.initial_introspection = encodedIntrospection;
+      requestBody.initial_introspection = introspection;
```

Comment:
the API doesn't complain for unexpected data, but here you're sending the interface name twice (as key and has value)

Reply:
You're right. I added another step where I make sure only wanted properties are cherry-picked